docs/src/main/asciidoc/security-ldap.adoc

```diff
@@ -0,0 +1,226 @@
+
+Let's start by implementing the `/api/public` endpoint. As you can see from the source code below, it is just a regular JAX-RS resource:
+
+[source,java]
+----
+package org.acme.elytron.security.ldap;
+
+import javax.annotation.security.PermitAll;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+@Path("/api/public")
+public class PublicResource {
+
+    @GET
+    @PermitAll
+    @Produces(MediaType.TEXT_PLAIN)
+    public String publicResource() {
+        return "public";
+   }
+}
+----
+
+The source code for the `/api/admin` endpoint is also very simple. The main difference here is that we are using a `@RolesAllowed` annotation to make sure that only users granted with the `adminRole` role can access the endpoint:
+
+
+[source,java]
+----
+package org.acme.elytron.security.ldap;
+
+import javax.annotation.security.RolesAllowed;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+@Path("/api/admin")
+public class AdminResource {
+
+    @GET
+    @RolesAllowed("adminRole")
+    @Produces(MediaType.TEXT_PLAIN)
+    public String adminResource() {
+         return "admin";
+    }
+}
+----
+
+Finally, let's consider the `/api/users/me` endpoint. As you can see from the source code below, we are trusting only users with the `standardRole` role.
+We are using `SecurityContext` to get access to the current authenticated Principal and we return the user's name. This information is loaded from the ldap server.
+
+[source,java]
+----
+package org.acme.elytron.security.ldap;
+
+import javax.annotation.security.RolesAllowed;
+import javax.inject.Inject;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.SecurityContext;
+
+@Path("/api/users")
+public class UserResource {
+
+    @GET
+    @RolesAllowed("standardRole")
+    @Path("/me")
+    @Produces(MediaType.APPLICATION_JSON)
+    public String me(@Context SecurityContext securityContext) {
+        return securityContext.getUserPrincipal().getName();
+    }
+}
+----
+
+=== Configuring the Application
+
+[source,properties]
+--
+quarkus.security.ldap.enabled=true
+
+quarkus.security.ldap.dir-context.principal=uid=tool,ou=accounts,o=YourCompany,c=DE
+quarkus.security.ldap.dir-context.url=ldaps://ldap.server.local
+quarkus.security.ldap.dir-context.password=PASSWORD
+
+quarkus.security.ldap.identity-mapping.rdn-identifier=uid
+quarkus.security.ldap.identity-mapping.search-base-dn=ou=users,ou=tool,o=YourCompany,c=DE
+
+quarkus.security.ldap.identity-mapping.attribute-mapping.from=cn
+quarkus.security.ldap.identity-mapping.attribute-mapping.to=groups
+quarkus.security.ldap.identity-mapping.attribute-mapping.filter=(member=uid={0})
+quarkus.security.ldap.identity-mapping.attribute-mapping.filter-base-dn=ou=roles,ou=tool,o=YourCompany,c=DE
```

Comment:
I'm surprised we can have only one `attribute-mapping`?

Reply:
Hmm you are right, one should be able to define multiple attribute-mappings. But currently the quarkus config only allows Map<String, AttributeMappingConfig> but not Map<Integer, AttributeMappingConfig> or List<AttributeMappingConfig>. For my usecase a single attribute-mapping was enough.

Reply:
Added possibility to define multiple attribute-mappings. Pushing soon.